docs/niscope/functions.rst

```diff
@@ -1669,12 +1669,12 @@ niscope.Session methods
 
 
 
-.. py:method:: read(num_samples, timeout=datetime.timedelta(seconds=5.0))
+.. py:method:: read(num_samples=None, relative_to=niscope.FetchRelativeTo.PRETRIGGER, offset=0, record_number=0, num_records=None, timeout=datetime.timedelta(seconds=5.0))
 
     Initiates an acquisition, waits for it to complete, and retrieves the
     data. The process is similar to calling :py:meth:`niscope.Session._initiate_acquisition`,
     :py:meth:`niscope.Session.acquisition_status`, and :py:meth:`niscope.Session._fetch`. The only difference is
-    that with :py:meth:`niscope.Session.read`, you enable all channels specified with
+    that with :py:meth:`niscope.Session._read`, you enable all channels specified with
```

Comment:
:(
Question: does this generate a build-time warning, like the ones we generate when the function isn't found at all?
I know we won't fix this for 1.0 but an automated way (i.e. build time warning) to get the full list would be a good thing.

Reply:
Separate.